core/src/main/java/org/elasticsearch/ingest/ConfigurationUtils.java

```diff
@@ -256,6 +257,11 @@ private static Processor readProcessor(ProcessorsRegistry processorRegistry, Str
             List<Processor> onFailureProcessors = readProcessorConfigs(onFailureProcessorConfigs, processorRegistry);
             Processor processor;
             processor = factory.create(config);
+
+            if (onFailureProcessorConfigs != null && onFailureProcessors.isEmpty() == true) {
```

Comment:
I prefer that as well, but for some reason remembered someone telling me to change it to `== true` since it is less error prone. I noticed it was used for config checking `config.isEmpty() == true`